### PR TITLE
[IMP] open_academy: Add many2one and one2many relation between course and session models T#52898

### PR DESCRIPTION
**Pull request by rolandojduartem**
**Course menu**
I updated the views to show the responsibles using many2one relations.
**_Tree view_**
Screenshot from 2022-01-11 19-05-26
**_Form view_**
Screenshot from 2022-01-11 19-05-33

**Session menu**
I updated the views to show the courses and instructors using many2one and one2many relations.
**_Tree view_**
Screenshot from 2022-01-11 19-05-42
**_Form view_**
Screenshot from 2022-01-11 19-05-47

### CODE REVIEW

**Comment by rolandojduartem** [comment]
@payen000 and @Jerudiel could you review this pull request, please?

**Review by payen000 — APPROVED** [review]
+1 Looks good to me! :)

**Comment by Jerudiel** [comment]
+1 looks good!

**Comment by rolandojduartem** [comment]
Hi, @frahikLV , could you review this PR?

File: open_academy/views/course.xml [hunk 1]
```diff
@@ -22,6 +23,7 @@
     <field name="model">course</field>
     <field name="arch" type="xml">
       <tree>
+        <field name="responsible_id"/>
```
Comment (frahikLV) [inline open_academy/views/course.xml:26]: Set the `responsible_id` after the `title`

File: open_academy/views/course.xml [hunk 2]
```diff
@@ -32,6 +34,7 @@
     <field name="model">course</field>
     <field name="arch" type="xml">
       <search>
+        <field name="responsible_id"/>
```
Comment (frahikLV) [inline open_academy/views/course.xml:37]: Set the `responsible_id` after the `title`

File: open_academy/views/course.xml [hunk 3]
```diff
@@ -6,10 +6,11 @@
       <form>
         <sheet>
           <group>
+            <field name="responsible_id"/>
```
Comment (frahikLV) [inline open_academy/views/course.xml:9]: Set the `responsible_id` after the `title`

File: open_academy/views/session.xml [hunk 4]
```diff
@@ -28,6 +30,8 @@
     <field name="model">session</field>
     <field name="arch" type="xml">
       <tree>
+        <field name="course_id"/>
+        <field name="instructor_id"/>
```
Comment (frahikLV) [inline open_academy/views/session.xml:34]: name field must be first that the `course_id` and `instructor_id`

File: open_academy/views/session.xml [hunk 5]
```diff
@@ -6,6 +6,8 @@
       <form>
         <sheet>
           <group>
+            <field name="course_id"/>
+            <field name="instructor_id"/>
```
Comment (frahikLV) [inline open_academy/views/session.xml:10]: name field must be first that the `course_id` and `instructor_id`

File: open_academy/views/session.xml [hunk 6]
```diff
@@ -40,6 +44,8 @@
     <field name="model">session</field>
     <field name="arch" type="xml">
       <search>
+        <field name="course_id"/>
+        <field name="instructor_id"/>
```
Comment (frahikLV) [inline open_academy/views/session.xml:48]: name field must be first that the `course_id` and `instructor_id`

File: open_academy/demo/session.xml [hunk 7]
```diff
@@ -1,11 +1,15 @@
 <odoo>
     <record model='session' id='session_demo1'>
+        <field name='course_id'>1</field>
+        <field name='instructor_id'>1</field>
```
Comment (frahikLV) [inline open_academy/demo/session.xml:4]: Never add "fixed" ids, use `ref` attribute, similar as:

https://github.com/odoo/odoo/blob/15.0/addons/account/data/account_data.xml#L147

File: open_academy/demo/session.xml [hunk 8]
```diff
@@ -1,11 +1,15 @@
...
         <field name='name'>Python</field>
         <field name='start_date'>2022-02-20</field>
         <field name='duration'>2.5</field>
         <field name='number_seat'>20</field>
       </record>
       <record model='session' id='session_demo2'>
+        <field name='instructor_id'>2</field>
+        <field name='course_id'>2</field>
```
Comment (frahikLV) [inline open_academy/demo/session.xml:12]: Never add "fixed" ids, use `ref` attribute, similar as:

https://github.com/odoo/odoo/blob/15.0/addons/account/data/account_data.xml#L147

File: open_academy/demo/course.xml [hunk 9]
```diff
@@ -1,9 +1,11 @@
 <odoo>
     <record model='course' id='course_demo1'>
+        <field name='responsible_id'>1</field>
         <field name='title'>Python</field>
         <field name='description'>Python Course</field>
       </record>
       <record model='course' id='course_demo2'>
+        <field name='responsible_id'>2</field>
```
Comment (frahikLV) [inline open_academy/demo/course.xml:8]: Never add "fixed" ids, use ref attribute, similar as:

https://github.com/odoo/odoo/blob/15.0/addons/account/data/account_data.xml#L147

File: open_academy/demo/course.xml [hunk 10]
```diff
@@ -1,9 +1,11 @@
 <odoo>
     <record model='course' id='course_demo1'>
+        <field name='responsible_id'>1</field>
```
Comment (frahikLV) [inline open_academy/demo/course.xml:3]: Never add "fixed" ids, use ref attribute, similar as:

https://github.com/odoo/odoo/blob/15.0/addons/account/data/account_data.xml#L147

File: open_academy/models/course.py [hunk 11]
```diff
@@ -4,6 +4,7 @@
 class Course(models.Model):
     _name = 'course'
     _description = 'Course'
+    identifier = fields.One2many('session', 'course_id')
```
Comment (frahikLV) [inline open_academy/models/course.py:7]: Why are you naming this field as `identifier`?, Please change it to `session_ids`

**Review by frahikLV — CHANGES_REQUESTED** [review]
@rolandojduartem can you address this changes?

**Comment by frahikLV** [comment]
@rolandojduartem 
Please rename your commits message:

From:
"[IMP] open_academy: Create relation between models T#52898"
To:
"[IMP] open_academy: Include relation between course with responsible and sessions with course and instructor T#52898"

It doesn't matter if the title it's too long

From: 
"[IMP] open_academy: Modify views to shows the new relations T#52898"
To:
[IMP] open_academy: Show course and instructor on session views and responsible on course views T#52898

From:
"[IMP] open_academy: Modify demo data with new columns T#52898"
To:
[IMP] open_academy: Include a responsible, course and instructor information on demo data T#52898

From:
"[IMP] open_academy: Add many2one and one2many relation between course and session models T#52898"
To:
"[IMP] open_academy: Include a session_ids o2m relation on course model T#52898"

**Comment by rolandojduartem** [comment]
It is ready, @frahikLV , could you review it again, please?

**Review by frahikLV — CHANGES_REQUESTED** [review]
@rolandojduartem can you address this changes?

**Comment by rolandojduartem** [comment]
@frahikLV , the change was applied.

File: open_academy/views/session.xml [hunk 12]
```diff
@@ -6,7 +6,9 @@
       <form>
         <sheet>
           <group>
-            <field name="name"/>
+             <field name="name"/>
```
[the remark this answers is not in the document]
Comment (rolandojduartem) [reply open_academy/views/session.xml:9]: @frahikLV resolved!

**Comment by frahikLV** [comment]
@rolandojduartem Great! I will merge this PR, ensure to rebase your new branch to get this changes

**Review by frahikLV — APPROVED** [review]
@rolandojduartem Great! I will merge this PR, ensure to rebase your new branch to get this changes

**Pull request merged**